### ISSUE TITLE
CLI "is not a valid version string" bug

### ISSUE THREAD

**Issue by olicooper** [abpframework/abp#5154]
Abp CLI v3.1.0-rc.1

I was updating the core system and also tried updating the Angular client from `v3.0.5`/`v3.1.0-preview20200813` to `v3.1.0-rc.1` Using `abp update` or `abp switch-to-preview` throws the error:

```
 One or more errors occurred. ('~3.0.5' is not a valid version string. (Parameter 'value'))
System.AggregateException: One or more errors occurred. ('~3.0.5' is not a valid version string. (Parameter 'value'))
 ---> System.ArgumentException: '~3.0.5' is not a valid version string. (Parameter 'value')
   at NuGet.Versioning.SemanticVersion.Parse(String value)
   at Volo.Abp.Cli.ProjectModification.NpmPackagesUpdater.TryUpdatingPackage(String filePath, JProperty package, Boolean includePreviews, Boolean includeReleaseCandidates, Boolean switchToStable) in D:\ci\Jenkins\workspace\abp-framework-release\abp\framework\src\Volo.Abp.Cli.Core\Volo\Abp\Cli\ProjectModification\NpmPackagesUpdater.cs:line 229
   at Volo.Abp.Cli.ProjectModification.NpmPackagesUpdater.UpdatePackagesInFile(String filePath, Boolean includePreviews, Boolean includeReleaseCandidates, Boolean switchToStable) in D:\ci\Jenkins\workspace\abp-framework-release\abp\framework\src\Volo.Abp.Cli.Core\Volo\Abp\Cli\ProjectModification\NpmPackagesUpdater.cs:line 198
   at Volo.Abp.Cli.ProjectModification.NpmPackagesUpdater.<>c__DisplayClass13_0.<<Update>g__UpdateAsync|0>d.MoveNext() in D:\ci\Jenkins\workspace\abp-framework-release\abp\framework\src\Volo.Abp.Cli.Core\Volo\Abp\Cli\ProjectModification\NpmPackagesUpdater.cs:line 61
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.WaitAllCore(Task[] tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Volo.Abp.Cli.ProjectModification.NpmPackagesUpdater.Update(String rootDirectory, Boolean includePreviews, Boolean includeReleaseCandidates, Boolean switchToStable) in D:\ci\Jenkins\workspace\abp-framework-release\abp\framework\src\Volo.Abp.Cli.Core\Volo\Abp\Cli\ProjectModification\NpmPackagesUpdater.cs:line 67
   at Volo.Abp.Cli.Commands.UpdateCommand.UpdateNpmPackages(String directory) in D:\ci\Jenkins\workspace\abp-framework-release\abp\framework\src\Volo.Abp.Cli.Core\Volo\Abp\Cli\Commands\UpdateCommand.cs:line 51
   at Volo.Abp.Cli.Commands.UpdateCommand.ExecuteAsync(CommandLineArgs commandLineArgs) in D:\ci\Jenkins\workspace\abp-framework-release\abp\framework\src\Volo.Abp.Cli.Core\Volo\Abp\Cli\Commands\UpdateCommand.cs:line 45
   at Volo.Abp.Cli.CliService.RunAsync(String[] args) in D:\ci\Jenkins\workspace\abp-framework-release\abp\framework\src\Volo.Abp.Cli.Core\Volo\Abp\Cli\CliService.cs:line 55
```

Some thoughts:
- Maybe it is because my projects were using the old preview system? (I have an `.npmrc` file with `@abp:registry=https://www.myget.org/F/abp-nightly/npm`)
- Maybe the tilde (~) needs stripping before parsing the version string? 
- Maybe the official `v3.0.5` release doesn't exist in the preview channel?

**Comment by olicooper** [comment]
When using `yarn add @abp/ng.core@3.1.0-rc.1` I get an error: `Couldn't find any versions for "@abp/ng.core" that matches "3.1.0-rc.1"`
But I can see it in npm: https://www.npmjs.com/package/@abp/ng.core/v/3.1.0-rc.1

**Comment by yekalkan** [comment]
Known issue. This will be fixed in 3.1.0-rc.2.
